blackduck/Client.py

```diff
@@ -257,6 +257,19 @@ def get_items(self, url, page_size=250, **kwargs):
 
             offset += page_size
 
+    def get_resource_by(self, field, value,  name, parent=None, **kwargs):
+        params = {
+            'q': [f"{field}:{value}"]
+        }
+        filtered = [i for i in self.get_resource(name, parent, params=params, **kwargs) if i.get(field) == value]
+        assert len(filtered) in [0,1], f"We either found the {field} or we didn't, but we should never find this many ({len(filtered)})"
+
+        return filtered[0] if filtered else None
+
+    def get_or_create_resource(self, field, value,  name, parent=None, **kwargs):
+        resource = self.get_resource_by(field, value, name, parent, **kwargs)
```

Comment:
Actually, looking at this again - think there should be more here.

Reply:
Uh, yeah, makes me wonder how my tests even worked. Investigating....

Reply:
some code got borked in transit. Fixing...

Reply:
Pushed an update with the (I think) now complete code. Still testing....